content/docs/2.0/concepts/external-scalers.md

```diff
@@ -0,0 +1,582 @@
+
+```json
+{
+  "name": "scaledobject-name",
+  "namespace": "scaledobject-namespace",
+  "scalerMetadata": {
+    "scalerAddress": "service-address.svc.local:9090",
+    "key1": "value1",
+    "key2": "value2"
+  }
+}
+```
+
+For `StreamingIsActive` KEDA establishes the connection to the GRPC server and expects `IsActive` events to be streamed as a response.
+
+## Implementing KEDA external scaler GRPC interface
+
+### Implementing an external scaler:
+
+#### 1. Download [`externalscaler.proto`](https://github.com/kedacore/keda/blob/v2/pkg/scalers/externalscaler/externalscaler.proto)
+
+#### 2. Prepare project:
+
+{{< collapsible "Golang" >}}
+
+2.1. Download [`./protoc`](https://github.com/protocolbuffers/protobuf/releases) for your platform
+
+2.2. get `protoc-gen-go`
+
+```bash
+go get github.com/golang/protobuf/protoc-gen-go@v1.3.2
+```
+
+2.3. Prepare project
+
+```bash
+go mod init example.com/external-scaler/sample
+mkdir externalscaler
+protoc externalscaler.proto --go_out=plugins=grpc:externalscaler
+```
+{{< /collapsible >}}
+
+{{< collapsible "C#" >}}
+2.1. Create a new project
+
+```bash
+dotnet new console -o ExternalScalerSample
+cd ExternalScalerSample
+
+# add Grpc.AspNetCore
+dotnet add package Grpc.AspNetCore
+dotnet add package Newtonsoft.Json
+
+# Create a Protos and Services folders
+mkdir Protos
+mkdir Services
+```
+
+2.2. Move `externalscaler.proto` to `Protos` folder
+
+2.3. Compile `externalscaler.proto` using this in `ExternalScalerSample.csproj`
+
+```xml
+  <ItemGroup>
+    <Protobuf Include="Protos\externalscaler.proto" GrpcServices="Server" />
+  </ItemGroup>
+```
+
+{{< /collapsible >}}
+
+{{< collapsible "Javascript" >}}
+2.1. Prepare the project
+
+```bash
+npm install --save grpc request
+```
+{{< /collapsible >}}
+
+<br />
+
+#### 3. Implementing `IsActive`
+
+Just like `IsActive(ctx context.Context) (bool, error)` in the go interface, `IsActive` method in the GRPC interface is called every `pollingInterval` with a `ScaledObjectRef` object that contains the scaledObject name, namespace, and scaler metadata. This section implements an external scaler that queries earthquakes from https://earthquake.usgs.gov/ and scales the deployment if there has been more than 2 earthquakes with `magnitude > 1.0` around a particular longitude/latitude in the previous day
+
+`ScaledObject`
+
+```yaml
+apiVersion: keda.sh/v1alpha1
+kind: ScaledObject
+metadata:
+  name: scaledobject-name
+  namespace: scaledobject-namespace
+spec:
+  scaleTargetRef:
+    name: deployment-name
+  triggers:
+    - type: external
+      metadata:
+        scalerAddress: earthquake-scaler:9090
+        longitude: "-122.335167"
+        latitude: "47.608013"
+```
+
+{{< collapsible "Golang" >}}
+
+Full implementation can be found here: https://github.com/ahmelsayed/external-scaler-samples
```

Comment:
Can we donate this to `kedacore`?

Reply:
yes, was just preparing the repo. Will move it there.

Reply:
Thanks and also for the repo, very useful to have a reference as well!